src/aleph/storage.py

```diff
@@ -194,8 +195,16 @@ async def get_hash_content(
 
         # Try to retrieve the data from the DB, then from the network or IPFS.
         content = await self.storage_engine.read(filename=content_hash)
+
         if content is not None:
-            source = ContentSource.DB
+            # check json and fix if corrupted
+            try:
+                json_content = aleph_json.loads(content)
+                source = ContentSource.DB
+            except orjson.JSONDecodeError as e:
+                LOGGER.warning("Can't decode JSON, Change source...")
+                await self.storage_engine.delete(filename=content_hash)
+                content = None
```

Comment:
I don't think you can assume that everything this function retrieves is going to be JSON. Pretty sure it should be treated as random binary data at this level. iirc there's a dedicated function that calls this and then deserializes the content to JSON, let me find the name.